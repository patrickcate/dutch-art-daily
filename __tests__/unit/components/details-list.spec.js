import DetailsList from '@components/details-list.vue'

describe('DetailsList Component', () => {
  let wrapper
  let options

  beforeEach(() => {
    options = {
      propsData: {
        id: '01-01',
      },
    }

    wrapper = createWrapper(DetailsList, options)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('does not render correctly', () => {
    const wrapperComponent = createWrapper(
      DetailsList,
      {
        ...options,
      },
      {
        getters: {
          getArtworkById: () => () => {
            return null
          },
        },
      }
    )

    expect(wrapperComponent).toMatchSnapshot()
  })
})
