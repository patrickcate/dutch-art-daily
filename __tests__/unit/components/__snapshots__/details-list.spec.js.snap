// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsList Component does not render correctly 1`] = ``;

exports[`DetailsList Component renders correctly 1`] = `
<ul class="details-list">
  <li class="details-list__item"><svg width="1em" height="1em" class="base-icon details-list__icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span><span class="u-sr-only">Date:</span> c. 1657–1658</span></li>
  <li class="details-list__item"><svg width="1em" height="1em" class="base-icon details-list__icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span><span class="u-sr-only">Medium:</span> Oil<span>
        on Canvas</span></span></li>
  <li class="details-list__item"><svg width="1em" height="1em" class="base-icon details-list__icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span><span class="u-sr-only">Size:</span> 45.5
    <abbr title="centimeters">cm</abbr> <span aria-hidden="true">×</span> <span class="u-sr-only">by</span> 41
    <abbr title="centimeters">cm</abbr></span>
  </li>
  <li class="details-list__item"><svg width="1em" height="1em" class="base-icon details-list__icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span><span class="u-sr-only">Location:</span> Rijksmuseum, Amsterdam
    </span></li>
  <li class="details-list__item"><svg width="1em" height="1em" class="base-icon details-list__icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span><span class="u-sr-only">Source:</span> Wikipedia:
    <a href="http://en.wikipedia.org/w/index.php?title=The_Milkmaid_(Vermeer)&amp;oldid=594234689" rel="noopener noreferrer external">
      The Milkmaid
    </a></span>
  </li>
</ul>
`;
